### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by loganek**
There's still more to be added here, but that's a start.

### CODE REVIEW

File: doc/adapters.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# WASM Runtime Adapters
+
+The test runner is designed to support different types of WebAssembly runtimes. Because the command line interface for WASM runtime is not standardized, every runtime can arbitrarily define how parameters should be passed to the runtime.
+
+WASI test runner is designed to support as many WASM runtimes as possible, therefore runtime peculiarities aren't hardcoded in the runner.
+
+In order to integrate WASM runtime with a test runner, the user has to provide a `runtime adapter`. It's an executable file that takes command-line arguments and translates them to a runtime call.
```
Comment (yamt) [inline doc/adapters.md:7]: do you mave any recommendations how to give misc configurations like the followings?
* runtime executable path
* aot compiler path
* runtime specific options to the above executables

**Comment by yamt** [comment]
i guess the expectation about WASI exit code should be documented here.

File: doc/adapters.md (hunk 1 above)
Comment (yamt) [reply doc/adapters.md:7]: i agree 1 is fine as far as our scope is simple wasi commands.

**Comment by loganek** [comment]
> i guess the expectation about WASI exit code should be documented here.

Agree, will add it to the adapter section.

**Review by cimacmillan — APPROVED** [review]

File: doc/adapters.md (hunk 1 above)
Comment (loganek) [reply doc/adapters.md:7]: Created a ticket for that: https://github.com/WebAssembly/wasi-testsuite/issues/19

**Pull request merged**